compression/lempel_ziv.py

```diff
@@ -0,0 +1,120 @@
+"""
+    One of the several implementations of Lempel–Ziv–Welch compression algorithm
+    https://en.wikipedia.org/wiki/Lempel%E2%80%93Ziv%E2%80%93Welch
+"""
+
+import os
+import sys
+import math
+
+
+def read_file_binary(file):
+    """
+    Reads given file as bytes and returns them as a long string
+    """
+    result = ""
+    try:
+        with open(file, "rb") as binary_file:
+            data = binary_file.read()
+            for dat in data:
+                curr_byte = "{0:08b}".format(dat)
+                result += curr_byte
+        return result
+    except IOError:
+        print("File not accessible")
+        sys.exit()
+
+
+def add_key_to_lexicon(lexicon, curr_string, index, last_match_id):
```

Comment:
Type hints?  Doctests?  See CONTRIBUTING.md.

Reply:
Got it! sorry for that, sure I'll add those but I have a question: is Doctest still needed for a method/function, that does not return a value? and if so, can you give me an example? (this function is one of the ones, that don't return anything) 
(def add_key_to_lexicon(lexicon: dict, curr_string: str, index: int, last_match_id: int) -> None:
this is the new header for it)

Reply:
```
>>> my_key in lexicon
False
>>> add_key_to_lexicon(lexicon, my_key)
>>> my_key in lexicon
True
```